### PR TITLE
Add -d:nimNoGetEntropy to allow running on macOS earlier than 10.12

### PR DESCRIPTION
**Pull request by iffy**
Related to #17370 but not the same. This change enables me to *compile* on a newer macOS a program that *runs* on an older macOS.

This does *not* restore the ability to compile on older macOS.

### CODE REVIEW

**Comment by Araq** [comment]
Why `nimNoGetEntropy`? Why not e.g. `nimOldOsX`?

**Comment by iffy** [comment]
> Why `nimNoGetEntropy`? Why not e.g. `nimOldOsX`?

I was trying to match `nimNoGetRandom` (which is adjacent to this) and be specific to this feature rather than possibly conflicting with a more general thing. I could imagine there being `nimOldOsX` now and `nimSlightlyNewerOldOsX` later :)  Ideally, I'd rather make the code just work for old and new, but that would require doing a runtime check rather than a compile-time check and I don't know how to do that.

**Comment by iffy** [comment]
Wait, I've got some idea how I could do it with `dynlib`. Let me try again.

**Comment by iffy** [comment]
Okay #20466 is the better fix.
